snmp/datadog_checks/snmp/snmp.py

```diff
@@ -311,6 +313,18 @@ def fetch_sysobject_oid(self, config):
         self.log.debug('Returned vars: %s', var_binds)
         return var_binds[0][1].prettyPrint()
 
+    def _profile_for_sysobject_oid(self, sys_object_oid):
+        """Return, if any, a matching profile for sys_object_oid.
+
+        If several profiles match, it will return the longer match, ie the
+        closest one to the sys_object_oid.
+        """
+        oids = [oid for oid in self.profiles_by_oid if fnmatch.fnmatch(sys_object_oid, oid)]
```

Comment:
In most integrations we use `regex` for matching, should we use it instead. (in that case, we probably need a config flag)

`fnmatch` seems simpler though for this case, especially because of the use of `.`.

Reply:
Yeah I think it this case it's good enough an less surprising. Any objection to keep using it?

Reply:
Sure, looks good to me. We can still implement regex later if there is a need (probably not) with a flag.